modules/router/src/forwarding.ts

```diff
@@ -156,11 +161,41 @@ export async function forwardTransferCreation(
       return Result.fail(
         new ForwardTransferError(ForwardTransferError.reasons.UnableToCalculateSwap, {
           message: swapRes.getError()?.message,
+          context: swapRes.getError()?.context,
         }),
       );
     }
     recipientAmount = swapRes.getValue();
-    logger.warn({ method, recipientAssetId, recipientAmount, recipientChainId }, "Inflight swap calculated");
+
+    // we need to get the proper transfer definition address
+    if (recipientChainId !== senderChainId) {
+      const senderTransferInfoRes = await chainReader.getRegisteredTransferByDefinition(
+        senderTransferDefinition,
+        senderChannel.networkContext.transferRegistryAddress,
+        senderChainId,
+      );
+      if (senderTransferInfoRes.isError) {
+        return Result.fail(
+          new ForwardTransferError(ForwardTransferError.reasons.InvalidTransferDefinition, {
+            message: swapRes.getError()?.message,
+          }),
+        );
+      }
+      const senderTransferInfo = senderTransferInfoRes.getValue();
+      recipientTransferDefinition = senderTransferInfo.name;
```

Comment:
It's misleading to call this the `transferDefinition`, since that always implies it has to be an onchain address. It does work as a `type`, but this should be renamed for clarity

Reply:
Good point. Fixed.